### ISSUE TITLE
Strange behaviour with catch-all alias

### ISSUE THREAD

**Issue by CupCakeArmy** [Mailu/Mailu#824]
**Version:** master

**Setup:**
- 1 domain (example.org) 
- 1 account (name@example.org)
- 1 catch-all alias for that domain (%@example.org)

**Problem:**
If I now add a second account (master@example.org) i cannot send or recieve messages with that account. I get an error stating that i don't have permissions to send from `master@example.org` and the received messages are passed to `name@example.org`.
Without the catch-all alias everything works fine.

I've see this ticket: https://github.com/Mailu/Mailu/issues/733
wich sounds basically identically, but it's already closed. If the changes there are implemented but not commited yet I'll close the ticket of course :)

Thanks for the great container! I was using mailcow before and it has a nice interface but it's really bloated and the backup does not work. Mailu is amazing! 🎉

**Comment by Nebukadneza** [comment]
Reproduced, nice find!

To me, it seems like this is a variety of #815, where aliases are ranked higher than users. This, at least to my understanding, leads to the user being overridden by an alias which he doesn’t own — hence he’s not allowed to send mail. To the same end, with `%@example.com` in place, no mail could ever reach `master@example.com` or `name@example.com`.

After #814 is in, I’ll open a PR for #815 , but according to my quick tests, that should fix your problem too.

**Comment by CupCakeArmy** [comment]
Yes, that is probably the exact problem. Looks like the catch-all are prioritised higher as the normal entries. I don't know how the catch all should affect sending an email, but then again I have absolutely no idea how smtp works so i'm trusting you guys.
You can close the issue if it's already tracked in #815 then :)

**Comment by Nebukadneza** [comment]
Thanks — doing that now and making #815 a bug instead ….
